src/server/service/kv.rs

```diff
@@ -783,10 +783,14 @@ impl<T: RaftStoreRouter<E::Local> + 'static, E: Engine, L: LockManager, Api: API
         let region_id = req.get_context().get_region_id();
         let (cb, f) = paired_future_callback();
         let mut split_keys = if req.is_raw_kv {
+            // TODO: process Txn & Raw data in a unified manner. CDC & BR have similar request.
             if !req.get_split_key().is_empty() {
-                vec![req.get_split_key().to_vec()]
+                vec![Api::encode_raw_key_owned(req.take_split_key(), None).into_encoded()]
             } else {
-                req.take_split_keys().to_vec()
+                req.take_split_keys()
```

Comment:
here, take_split_keys should be empty. So could it just return an empty vec?

Reply:
`split_key` and `split_keys` are two different fields, the former is the old fashion to split region.